docs/pages/docs/routing/middleware.mdx

```diff
@@ -549,29 +549,113 @@ export const config = {
 
 ### Example: Integrating with Supabase Authentication
 
-[`@supabase/auth-helpers-nextjs`](https://supabase.com/docs/guides/auth/auth-helpers/nextjs?language=ts#managing-session-with-middleware) exports `createMiddlewareClient` that accepts a `req` and `res` object. We can pass the incoming `req` and the `res` created with `createIntlMiddleware` to it.
+[`@supabase/ssr`](https://supabase.com/docs/guides/auth/server-side/creating-a-client) exports a `createServerClient` function. First we're going to create a utils function that invokes the `createServerClient` function and returns the `supabase` object.
+
+```ts filename="utils/supabase/middleware.ts"
+import {createServerClient, type CookieOptions} from '@supabase/ssr';
+import {type NextRequest, NextResponse} from 'next/server';
+
+export const createClient = (request: NextRequest) => {
+  // Create an unmodified response
+  let response = NextResponse.next({
+    request: {
+      headers: request.headers
+    }
+  });
+
+  const supabase = createServerClient(
+    process.env.NEXT_PUBLIC_SUPABASE_URL!,
+    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
+    {
+      cookies: {
+        get(name: string) {
+          return request.cookies.get(name)?.value;
+        },
+        set(name: string, value: string, options: CookieOptions) {
+          // If the cookie is updated, update the cookies for the request and response
+          request.cookies.set({
+            name,
+            value,
+            ...options
+          });
+          response = NextResponse.next({
```

Comment:
This looks a bit suspicious to me. I saw it's mentioned in the Supabase docs too, but I'm wondering why this is necessary and also if it could break something. Do you think we can remove this?

Reply:
tried without, is working in development as well as in production, so I'd say lets keep it out